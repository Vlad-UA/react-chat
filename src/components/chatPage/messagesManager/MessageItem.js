import React from 'react';
import classNames from 'classnames';

import {withStyles} from 'material-ui/styles';
import {ListItem} from 'material-ui/List';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';

import AvatarItem from "../common/Avatar";

const styles = theme => ({
  root:{
    width: '100%',
  },
  paper: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    minWidth: '10%',
    maxWidth: '70%',
  }),

  myMessageListItem: {
    display: 'flex',
    flexDirection: 'row-reverse',
  },

  myMessagePaper: {
    backgroundColor: '#C7EDFC',
  },
});

const MessageItem = ({classes, sender, content}) => {
  const isMessageFromMe = sender === "me";

  return (
    <ListItem className={classNames(classes.root, isMessageFromMe && classes.myMessageListItem)}>
      <AvatarItem title={sender} lettersQuantity={1} colorFrom={sender}/>
      <Paper className={classNames(classes.paper, isMessageFromMe && classes.myMessagePaper)} elevation={4}>
        <Typography component="p" variant="caption">
          {sender}
        </Typography>
        <Typography variant="body1" component="p">
          {content}
        </Typography>
      </Paper>
    </ListItem>
  )
};

export default withStyles(styles)(MessageItem);
