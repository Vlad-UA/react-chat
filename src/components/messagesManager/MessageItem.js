import React from 'react';
import {withStyles} from 'material-ui/styles';
import classNames from 'classnames';

import {ListItem} from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';

import titleInitial from '../../utils/title-initials';

const styles = theme => ({
  paper: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
  }),

  myMessageListItem: {
    display: 'flex',
    flexDirection: 'row-reverse',
  },

  myMessagePaper: {
    backgroundColor: '#C7EDFC',
  },
});

class MessageItem extends React.Component {
  render() {
    const {classes, message} = this.props;

    const isMessageFromMe = message.sender === "me";

    return (
      <ListItem className={classNames(isMessageFromMe && classes.myMessageListItem)}>
        <Avatar>
          {titleInitial({title: message.sender, lettersQuantity: 1})}
        </Avatar>
        <Paper className={classNames(classes.paper, isMessageFromMe && classes.myMessagePaper)} elevation={4}>
          <Typography component="p">
            {message.sender}
          </Typography>
          <Typography variant="headline" component="p">
            {message.content}
          </Typography>
        </Paper>
      </ListItem>
    )
  }
}

export default withStyles(styles)(MessageItem);
