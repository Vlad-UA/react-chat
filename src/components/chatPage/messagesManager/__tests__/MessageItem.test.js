/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageItem from '../MessageItem';

const mockProps = {
  sender: {
    _id: '_id',
  },
  content: 'content',
  activeUser: {
    _id: '_id',
  },
  statusMessage: false,
  createdAt: '2018-03-16T10:53:23.200Z',
};

jest.mock('../../common/Avatar', () => () => 'Avatar');

describe('<MessageItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageItem {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot renders correctly - base message (not status message)', () => {
    const tree = renderer.create(<MessageItem {...mockProps} statusMessage={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - status message', () => {
    const tree = renderer.create(<MessageItem {...mockProps} statusMessage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
