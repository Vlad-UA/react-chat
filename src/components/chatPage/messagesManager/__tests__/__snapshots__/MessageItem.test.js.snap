// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageItem /> snapshot renders correctly - base message (not status message) 1`] = `
<li
  className="MuiListItem-root-74 MuiListItem-default-77 MuiListItem-gutters-81 MessageItem-root-68 MessageItem-myMessageListItem-70"
  disabled={false}
>
  Avatar
  <div
    className="MuiPaper-root-84 MuiPaper-elevation4-90 MuiPaper-rounded-85 MessageItem-paper-69 MessageItem-myMessagePaper-71"
  >
    <p
      className="MuiTypography-root-111 MuiTypography-caption-121"
    />
    <p
      className="MuiTypography-root-111 MuiTypography-body1-120"
    >
      content
    </p>
  </div>
</li>
`;

exports[`<MessageItem /> snapshot renders correctly - status message 1`] = `
<div
  className="MessageItem-root-68"
>
  <p
    className="MuiTypography-root-111 MuiTypography-body1-120 MessageItem-statusMessage-72"
  >
    <span
      className="MuiTypography-root-111 MuiTypography-caption-121 MessageItem-statusMessageUser-73"
      style={
        Object {
          "color": "#e91e63",
        }
      }
    />
    content
    <span
      className="MuiTypography-root-111 MuiTypography-caption-121"
    >
      3 months ago
    </span>
  </p>
</div>
`;
